src/LayerEditor/ui/diagram_editor/diagram.py

```diff
@@ -0,0 +1,365 @@
+from PyQt5 import QtWidgets, QtCore, QtGui
+from PyQt5.QtCore import Qt
+
+from LayerEditor.ui.data.regions import Regions
+from LayerEditor.ui.tools.cursor import Cursor
+from LayerEditor.ui.tools.selection import Selection
+
+
+from bgem.polygons.polygons import PolygonDecomposition
+from bgem.external import undo
+
+from LayerEditor.ui.diagram_editor.graphics_items.gs_point import GsPoint
+from LayerEditor.ui.diagram_editor.graphics_items.gs_polygon import GsPolygon
+from LayerEditor.ui.diagram_editor.graphics_items.gs_segment import GsSegment
+
+
+class Diagram(QtWidgets.QGraphicsScene):
+    selection_changed = QtCore.pyqtSignal()
+    # selection has changed
+
+    regionsUpdateRequired = QtCore.pyqtSignal()
+
+    def __init__(self, parent):
+        rect = QtCore.QRectF(-622500, 1128600, 400, 500)
+        super().__init__(rect, parent)
+        self.points = {}
+        self.segments = {}
+        self.polygons = {}
+
+        self.regions = Regions(self)
+
+        self.last_point = None
+        self.aux_pt, self.aux_seg = self.create_aux_segment()
+        self.hide_aux_line()
+
+        self._zoom_value = 1.0
+        self.selection = Selection(self)
```

Comment:
Is selection specific to single diagram or is it independent of the diagram? I think it is the second case, so the selection should probably not be part of the diagram.

Reply:
Solved in branch TB_LE_inicialization_and_IO